### PR TITLE
fix(sandbox): scrub dynamic credential env vars

### LINKED ISSUE #676: Sandbox env scrub misses custom apiKeyEnv and dynamic OAuth client secrets
## Summary

`scrubSensitiveEnv` removes a static secret list plus `AuthEnvVars` from the built-in provider catalog. Zero also supports secret environment-variable names outside that catalog:

- arbitrary provider-profile `apiKeyEnv` values, such as `COMPANY_LLM_SECRET`
- dynamic `ZERO_OAUTH_<NAME>_CLIENT_SECRET` variables

Those values remain inherited by sandboxed commands.

## Relevant code

- `internal/sandbox/runner.go`: `scrubSensitiveEnv` derives keys only from a fixed list and `providercatalog.All()`.
- `internal/config/types.go`: `ProviderProfile.APIKeyEnv` accepts a configured variable name.
- `internal/config/resolver.go`: the configured `APIKeyEnv` is read from the process environment.
- `internal/oauth/providers.go`: `envKey(name, "CLIENT_SECRET")` constructs secret names dynamically.

Observed at commit `6fc1220`.

## Reproduction

1. Configure a custom provider with `"apiKeyEnv": "COMPANY_LLM_SECRET"`.
2. Start Zero with `COMPANY_LLM_SECRET=example-secret`.
3. Run a sandboxed command that prints it.
4. The value is present because it is absent from the catalog-derived scrub list.

## Suggested fix

Pass resolved sensitive env names from configuration into sandbox construction and scrub dynamic secret namespaces such as `ZERO_OAUTH_*_CLIENT_SECRET`. Add tests with a non-catalog `apiKeyEnv` and custom OAuth provider name.

### PR DESCRIPTION
**Pull request by PierrunoYT**
## Summary

- pass every resolved provider profile's custom `apiKeyEnv` into the sandbox engine
- scrub those config-derived names in Linux, macOS, and Windows sandbox command plans
- scrub case-insensitive `ZERO_OAUTH_<NAME>_CLIENT_SECRET` variables for dynamically named OAuth providers
- keep the existing catalog-derived and static secret scrubbing behavior centralized

## Root cause

The sandbox environment scrubber knew only a static secret list and the built-in provider catalog. Custom provider profiles can name arbitrary credential variables through `apiKeyEnv`, and OAuth provider names create client-secret variables dynamically, so neither category was represented in that fixed set.

Sandboxed commands could therefore inherit those credentials even though built-in provider secrets were removed.

Fixes #676

## Validation

- `go fmt ./...`
- `go vet ./...`
- `go test ./internal/sandbox ./internal/cli -count=1`
- repository-wide `go test ./... -count=1` completed successfully for the changed packages and all but unrelated Windows process-lifecycle tests
- `go run golang.org/x/vuln/cmd/govulncheck@v1.3.0 ./...`: no vulnerabilities found
- pinned golangci-lint reported the same 35 pre-existing findings tracked by #680 and none in this patch

## Known baseline/environment limitations

- `TestLoadProviderCommandTimeout` takes about 10 seconds instead of its expected 5 seconds on this host.
- `TestManagerCheckRealGopls` fails because the installed `gopls` process exits with EOF.
- Several `internal/tools` process-termination and temporary-directory cleanup tests fail on Windows because child processes retain file handles.
- `make lint` uses a Unix shell assignment in `fmt-check` that does not run under this Windows shell.

## Summary by CodeRabbit

* **Security Improvements**
  * Sandbox commands now scrub additional provider credential environment variables using resolved configuration, consistently across Linux, macOS, and Windows.
  * Improved handling of dynamically named OAuth client-secret environment variables (case-insensitive `ZERO_OAUTH_*_CLIENT_SECRET` pattern).

* **Bug Fixes**
  * Fixed cases where sensitive configured keys were not removed in all execution paths, including degraded fallback scenarios.

* **Tests**
  * Added regression coverage to verify configured and pattern-based sensitive environment scrubbing behavior.

### CODE REVIEW

**Review by jatmn — CHANGES_REQUESTED** [review]
I found issues that need to be addressed before this is ready.

## Findings

- [P1] Scrub credentials on degraded command plans too
  `internal/sandbox/runner.go:169`
  `BuildCommandPlan` attaches the configured keys to `spec`, but the supported unavailable-backend fallback returns `directCommandPlan`, which simply leaves `Env` nil. The normal bash tool creates a spec with nil `Env`, so `exec.Cmd` inherits `COMPANY_LLM_SECRET` and `ZERO_OAUTH_<NAME>_CLIENT_SECRET` unchanged whenever the native backend is unavailable (the intended `EnforcementDegraded` path covered by `TestBuildCommandPlanDegradesUnavailableFallback` and `TestUnavailableBackendsDegradeForTargetPlatforms`). That leaves the reported credential-exfiltration scenario open on fallback hosts; construct a scrubbed environment for direct/degraded plans as well and add an inherited-environment regression test.

**Review by anandh8x — APPROVED** [review]
Re-reviewing against commit `7b59fc83` (head). The `nil `spec.Env` snapshot fix (patch 3/3) closes a real leak: direct/unwrapped command plans were inheriting the parent env unscrubbed, defeating the whole point of the scrubber — a sandboxed command could see `OPENAI_API_KEY` etc. by going through the direct plan path. That was the P1 and the fix is correct (snapshot `os.Environ()` then run `scrubSensitiveEnv` before forking). The configurable `SensitiveEnvKeys` plumbing from CLI → engine is clean, and the case-insensitive `ZERO_OAUTH_*_CLIENT_SECRET` pattern via `strings.ToUpper` is the right shape for env vars.

LGTM. Minor follow-up (not blocking): the `isDynamicSensitiveEnvKey` length check rejects the empty-provider case `ZERO_OAUTH__CLIENT_SECRET` (double underscore) — a one-line test for that edge case would be cheap. Also worth documenting the dynamic-pattern list near the function so a future maintainer knows where to add a second pattern.

Cross-PR note: #682 sits between #681 and #685. Recommend rebasing #682 onto #681, then #685 onto #682, to resolve the `credentialDenyReadPaths` / `scrubSensitiveEnv` signature overlap cleanly.

**Review by gnanam1990 — COMMENTED** [review]
Local review: built and ran `go test ./internal/cli ./internal/sandbox` on darwin/arm64 (cli 94s); all pass. The direct/degraded-plan env scrubbing (including the `spec.Env == nil` -> `os.Environ()` snapshot) is a genuine security fix. One integration note.

File: internal/sandbox/runner.go [hunk 1]
```diff
@@ -967,7 +992,7 @@ func regexpQuoteMeta(value string) string {
 	return replacer.Replace(value)
 }
 
-func scrubSensitiveEnv(env []string) []string {
+func scrubSensitiveEnv(env []string, additionalKeys ...string) []string {
```
Comment (gnanam1990) [inline internal/sandbox/runner.go:995]: [P2] Sibling-PR merge conflict on `scrubSensitiveEnv` in `runner.go`

This PR changes the signature to `scrubSensitiveEnv(env, additionalKeys ...string)` and adds `normalizeSensitiveEnvKeys`. #685 adds a hardcoded `ZERO_DAEMON_REMOTE_TOKEN_FILE` to the same function's literal list, and both touch `runner_test.go`'s `TestScrubSensitiveEnv`. Each merges into `main` cleanly on its own, but they conflict with each other; #681 also rewrites `credentialDenyReadPathsIn` in `profile.go`. Sequence the three sandbox credential PRs and rebase as needed.

**Review by jatmn — APPROVED** [review]
LGTM

**Review by Vasanthdev2004 — APPROVED** [review]
Looks good to me. This closes a real leak: config-derived apiKeyEnv names and dynamic ZERO_OAUTH_*_CLIENT_SECRET vars are now scrubbed on every plan path (linux/seatbelt/windows wrapped plus the degraded direct fallback that previously inherited the caller env unscrubbed). I checked out the branch and confirmed go build, go vet, and gofmt are clean on the touched files, and the new tests (TestScrubSensitiveEnv, TestEngineScrubsConfiguredSensitiveEnvKeys, TestBuildCommandPlanDegradedFallbackScrubsInheritedEnv, TestProviderSensitiveEnvKeys) pass. The bare ZERO_OAUTH_CLIENT_SECRET (no provider segment) the reviewer noted is preserved by the prefix/suffix length guard, but that's fine in practice: Zero's envKey always emits ZERO_OAUTH_<NAME>_CLIENT_SECRET with a name segment, so that exact bare name is never a credential Zero sets.

**Review by kevincodex1 — COMMENTED** [review]
lgtm
